web/src/app/util/NumberField.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from 'react'
+import { TextField, TextFieldProps } from '@material-ui/core'
+
+type NumberFieldProps = TextFieldProps & {
+  // float indicates that decimals should be accepted
+  float?: boolean
+  min?: number
+  max?: number
+  value: string
+  onChange: (val: string) => void
```

Comment:
I was following the props from `@material-ui/core/TextField` where value is listed as always returning as a string. If we followed their docs explicitly, we would need to pass `e` as an object instead of `e.target.value`. I think both should work, but the clear message of "I am updating a string value" is why I went with the current impl

![image](https://user-images.githubusercontent.com/11381794/96154733-fc587800-0ec3-11eb-9ee5-2470af7a481f.png)

Reply:
Right, since we're manipulating the target value for the `onChange` we wouldn't be able to do that without cloning the event and returning it with the new value.

Reply:
We should be able to do that with `e.target.value = newValue` and then call `onChange(e)` right?

Reply:
Yeah I suppose we could do that, but is it proper to update a const that we get back from a response and then return it? We are handling both types within `FormField`: `(L147) let getValueOf = (e) => (e && e.target ? e.target.value : e)`